src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/SemanticTokensRangeEndpoint.cs

```diff
@@ -1,43 +1,30 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
-using System;
-using System.Diagnostics;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Razor.LanguageServer.Common;
 using Microsoft.AspNetCore.Razor.LanguageServer.EndpointContracts;
-using Microsoft.AspNetCore.Razor.Telemetry;
 using Microsoft.CommonLanguageServerProtocol.Framework;
-using Microsoft.Extensions.Logging;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic;
 
-[LanguageServerEndpoint(LspEndpointName)]
-internal sealed class SemanticTokensRangeEndpoint : IRazorRequestHandler<SemanticTokensRangeParams, SemanticTokens?>, ICapabilitiesProvider
+[LanguageServerEndpoint(Methods.TextDocumentSemanticTokensRangeName)]
+internal sealed class SemanticTokensRangeEndpoint(
+    IRazorSemanticTokensInfoService semanticTokensInfoService,
+    RazorLSPOptionsMonitor razorLSPOptionsMonitor,
+    IClientConnection clientConnection)
+    : IRazorRequestHandler<SemanticTokensRangeParams, SemanticTokens?>, ICapabilitiesProvider
 {
-    public const string LspEndpointName = Methods.TextDocumentSemanticTokensRangeName;
-    private RazorSemanticTokensLegend? _razorSemanticTokensLegend;
-    private readonly ITelemetryReporter? _telemetryReporter;
-
-    public SemanticTokensRangeEndpoint(ITelemetryReporter? telemetryReporter)
-    {
-        _telemetryReporter = telemetryReporter;
-    }
+    private readonly IRazorSemanticTokensInfoService _semanticTokensInfoService = semanticTokensInfoService;
```

Comment:
>_semanticTokensInfoService [](http://example.com/codeflow?start=53&length=26)

Why the field instead of using primary constructor param in the methods?

Reply:
See reply above, but TL;DR I like the underscore :)

Reply:
I like the underscores too :) Also see my response above. I don't have strong feeling either way, other than - do the primary constructors even make sense then?